packages/babel-plugin-transform-react-jsx/src/transform-classic.js

```diff
@@ -46,29 +53,25 @@ export default declare((api, options) => {
 
       let pragma = PRAGMA_DEFAULT;
       let pragmaFrag = PRAGMA_FRAG_DEFAULT;
-      let pragmaSet = !!options.pragma;
-      let pragmaFragSet = !!options.pragmaFrag;
 
       if (file.ast.comments) {
         for (const comment of (file.ast.comments: Array<Object>)) {
           const jsxMatches = JSX_ANNOTATION_REGEX.exec(comment.value);
           if (jsxMatches) {
             pragma = jsxMatches[1];
-            pragmaSet = true;
           }
           const jsxFragMatches = JSX_FRAG_ANNOTATION_REGEX.exec(comment.value);
           if (jsxFragMatches) {
             pragmaFrag = jsxFragMatches[1];
-            pragmaFragSet = true;
           }
         }
       }
 
       state.set("jsxIdentifier", createIdentifierParser(pragma));
       state.set("jsxFragIdentifier", createIdentifierParser(pragmaFrag));
       state.set("usedFragment", false);
-      state.set("pragmaSet", pragmaSet);
-      state.set("pragmaFragSet", pragmaFragSet);
+      state.set("pragmaSet", pragma !== DEFAULT.pragma);
+      state.set("pragmaFragSet", pragmaFrag !== DEFAULT.pragmaFrag);
```

Comment:
This part causes an issue when users want to use custom `pragma` with default `pragmaFrag` (namely the `React.Fragment`).
This is the case in the [emotion babel preset](https://github.com/emotion-js/emotion/tree/master/packages/babel-preset-css-prop), and causes a problem in [gatsby](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-emotion/src/gatsby-node.js) too